### PR TITLE
First integration of Pixel Alignment of large structures in PCL (76X)

### PR DESCRIPTION
**Pull request by cerminar**
This PR integrates the alignment of the pixel large structures as a
new PCL workflow for Tier0.
There is no code change to any production workflow, only the alca
configurations are affected by the changes.

The new workflow has been integrated in the relval 1001 (which now has
one more step).

### CODE REVIEW

**Comment by diguida** [comment]
please test

**Comment by cerminar** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by cerminar** [comment]
hello, it would be very useful to merge this asap not to loose all this work. Can the relevant parties sign unless there is something wrong?
Thanks.

**Comment by boudoul** [comment]
@cmsbuild , please test

**Comment by boudoul** [comment]
+1

**Comment by diguida** [comment]
@davidlange6 
Unfortunately @ggovi is not available.
When we signed this PR, we also gave a look to the Condition object changes. There is only an addition of a new class `FileBlobCollection`, actually used as a Data class (stored in the Event Tree), and the corresponding update of the ROOT dictionaries. They all look good to me. Only one doubt: these lines are added in the `classes_def.xml` file:

````
<class name="FileBlobCollection" class_version="0"/>
<class name="edm::Wrapper<FileBlobCollection>" class_version="0"/>
````

There I do not know if class version 0 is accepted by the data format classes.

**Comment by diguida** [comment]
From a quick look to class dictionaries in DataFormats, the reflection definition should look a bit different...
@davidlange6 can you please advise?

**Comment by diguida** [comment]
See, e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_X/DataFormats/BeamSpot/src/classes_def.xml

**Comment by Dr15Jones** [comment]
My recollection is ROOT used class version 0 to mean never store.

**Comment by diguida** [comment]
@Dr15Jones thanks for the check.
Would you therefore suggest to use the same convention as Data classes, i.e.

````
  <class name="FileBlobCollection" ClassVersion="10">
   <version ClassVersion="10" checksum="...."/>
  </class>
````

**Comment by Dr15Jones** [comment]
I suggest using 2 since this is a new class

Chris

**Comment by ggovi** [comment]
+1

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc [hunk 1]
```diff
@@ -0,0 +1,1247 @@
...
+void PCLTrackerAlProducer
+::endRun(const edm::Run& run, const edm::EventSetup& setup)
+{
+  // TODO: Either MP nor HIP is implementing the endRun() method... so this
+  //       seems to be useless?
+
+  if (tkLasBeamTag_.encode().size()) {
+    edm::Handle<TkFittedLasBeamCollection> lasBeams;
+    edm::Handle<TsosVectorCollection> tsoses;
+    run.getByLabel(tkLasBeamTag_, lasBeams);
```
Comment (davidlange6) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @cerminar - needs consumes migration (and ideally getByToken)

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc [hunk 2]
```diff
@@ -0,0 +1,1247 @@
...
+  ++nevent_;
+
+  // reading in survey records
+  readInSurveyRcds(setup);
+
+  // Retrieve trajectories and tracks from the event
+  // -> merely skip if collection is empty
+  edm::Handle<TrajTrackAssociationCollection> handleTrajTracksCollection;
+
+  if (event.getByLabel(tjTkAssociationMapTag_, handleTrajTracksCollection)) {
```
Comment (davidlange6) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: same here - consumes

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc (hunk 1 above)
Comment (cerminar) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @DirtyDan88 please could you have a look into this? Thanks,
G

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc (hunk 2 above)
Comment (cerminar) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @DirtyDan88 ditto.

**Comment by cerminar** [comment]
Superseded by  #10971

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc (hunk 1 above)
Comment (DirtyDan88) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @cerminar @davidlange6 Sorry, there is a type in the TODO comment, should be 'neither'... the method call in line 198 (and 203) is useless, because neither MP nor HIP implements this method, so also this method seems to be useless.

File: Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc (hunk 2 above)
Comment (DirtyDan88) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @cerminar As I see this is already done in another commit... is there a new PR for this commit?

Comment (mmusich) [inline Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc]: @DirtyDan88 chekout https://github.com/cms-sw/cmssw/pull/10971
